### PR TITLE
Probe debug output goes into CDC with timestamp

### PR DESCRIPTION
**Pull request by rgrr**
I need some debug output from the probe itself, because I'm currently working on it.

And because I have no serial/USB adapter (ok, I could have used another Pico for this...) idea was to use CDC for probe debug output.  Mixing target debug output with probe does not harm TMO, because for regular users it can be switched off completely.

For easier measurements I also included a time stamp for the debug output.

### CODE REVIEW

File: src/cdc_uart.c [hunk 1]
```diff
@@ -37,36 +38,63 @@ TickType_t last_wake, interval = 100;
...
-        int written = 0;
-        /* Implicit overflow if we don't write all the bytes to the host.
-         * Also throw away bytes if we can't write... */
-        if (rx_len) {
-          written = MIN(tud_cdc_write_available(), rx_len);
-          if (written > 0) {
-            tud_cdc_write(rx_buf, written);
-            tud_cdc_write_flush();
-          }
+      was_connected = 1;
```
Comment (lurch) [inline src/cdc_uart.c:69]: Probably better to stick to 4-spaces per indent? (and yes, the original code wasn't entirely consistent here)

Comment (rgrr) [reply src/cdc_uart.c:69]: Do you suggest some reformatting?  What are the actual rules?

Comment (lurch) [reply src/cdc_uart.c:69]: I dunno if there are any hard-and-fast rules, but https://github.com/raspberrypi/pico-sdk should probably be considered as the default coding style? :shrug:

Comment (rgrr) [reply src/cdc_uart.c:69]: seems to be a little bit like "WebKit".

**Comment by rgrr** [comment]
> While this may be useful for probe development, I'm not sure this implementation belongs in this repo. It won't help if you're wanting to debug the CDC UART itself, or if you want to turn on tinyUSB debug messages. It will also prevent the use of the UART by any host software that e.g. expects framed or formatted messages.
> 
> Can you instantiate a second cdc interface and route the debug messages there?

Yes, this feature is intended for probe development.  Release firmwares could switch this completely off by proper definitions in picoprobe_config.h so the target operation will not be influenced.

I have to admit that this is kind of lazy implementation but if you insist, I will do my best implementing it with a second CDC interface.

**Comment by rgrr** [comment]
Ok, I have committed a version which sends debug output to a secondary CDC if compiled for Debug.  If compiled for Release there will be only one CDC as before

**Comment by P33M** [comment]
If you still want this merged then please rebase on latest master.

**Comment by rgrr** [comment]
I'm closing it.  Thanks for your work anyway.

**Pull request closed without merging**